### PR TITLE
fix view bug

### PR DESCRIPTION
**Pull request by lsy3993**

### CODE REVIEW

File: pkg/ccr/job.go [hunk 1]
```diff
@@ -946,6 +946,24 @@ func (j *Job) handleCreateTable(binlog *festruct.TBinlog) error {
...
+		log.Debugf("create view, use dest db name to replace source db name")
+
+		// replace `internal`.`source_db_name`. to `internal`.`dest_db_name`.
+		originalName := "`internal`.`" + strings.TrimSpace(j.Src.Database) + "`."
+		dbNameRegex := regexp.MustCompile(originalName)
+		replaceName := "`internal`.`" + strings.TrimSpace(j.Dest.Database) + "`."
+		createTable.Sql = dbNameRegex.ReplaceAllString(createSql, replaceName)
+		log.Debugf("original create view sql is %s, after repalce, now sql is %s", createSql, createTable.Sql)
+	}
+
```
Comment (w41ter) [inline pkg/ccr/job.go]: Move SQL related operations into Spec.go:CreateTable, and add a parameter 'srcDatabase'.

```go
createTable.Sql = dbNameRegex.ReplaceAllString(createSql, replaceName)
```

`strings.ReplaceAll` is enough here.

File: pkg/ccr/base/spec.go [hunk 2]
```diff
@@ -297,6 +297,67 @@ func (s *Spec) GetAllTables() ([]string, error) {
...
+
+	return results, nil
+}
+
+func (s *Spec) GetAllViewsFromTable(tableName string) ([]string, error) {
+	log.Debugf("get all view from table %s", tableName)
+
+	var results []string
+	// first, query information_schema.tables with table_schema and table_type, get all views' name
+	querySql := fmt.Sprintf("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s' AND  table_type = 'VIEW'", s.Database)
```
Comment (w41ter) [inline pkg/ccr/base/spec.go:333]: `desc ${tableName} all` might be easier and more efficient than this.

See https://doris.apache.org/docs/1.2/query-acceleration/materialized-view/#query-materialized-views.

File: pkg/ccr/job.go [hunk 3]
```diff
@@ -1034,14 +1052,31 @@ func (j *Job) handleAlterJob(binlog *festruct.TBinlog) error {
...
-		var destTableName string
-		if j.SyncType == TableSync {
-			destTableName = j.Dest.Table
-		} else {
-			destTableName = alterJob.TableName
+		// before drop table, drop related view firstly
+		if !allViewDeleted {
+			views, err := j.IDest.GetAllViewsFromTable(destTableName)
+			if err != nil {
+				return err
```
Comment (w41ter) [inline pkg/ccr/job.go]: Logging errors and breaking here instead of returning, if you aim to loop?

**Review by w41ter — CHANGES_REQUESTED** [review]

File: pkg/ccr/base/spec.go (hunk 2 above)
Comment (lsy3993) [reply pkg/ccr/base/spec.go:333]: 'desc ${tableName} all' don't support common view, materialized view is supported

File: pkg/ccr/job.go (hunk 1 above)
Comment (lsy3993) [reply pkg/ccr/job.go]: done

File: pkg/ccr/job.go (hunk 3 above)
Comment (lsy3993) [reply pkg/ccr/job.go]: done

File: pkg/ccr/job.go [hunk 4]
```diff
@@ -1034,14 +1034,42 @@ func (j *Job) handleAlterJob(binlog *festruct.TBinlog) error {
...
+
+			var dropViewFailed bool = false
+			for _, view := range views {
+				if err := j.IDest.DropView(view); err != nil {
+					log.Errorf("when alter job, drop view %s failed, err : %v", view, err)
+					dropViewFailed = true
+				}
+			}
+			if dropViewFailed {
+				break
```
Comment (w41ter) [inline pkg/ccr/job.go]:
```suggestion
				continue
```

File: pkg/ccr/job.go [hunk 5]
```diff
@@ -1034,14 +1034,42 @@ func (j *Job) handleAlterJob(binlog *festruct.TBinlog) error {
...
 	for {
-		// drop table dropTableSql
-		var destTableName string
-		if j.SyncType == TableSync {
-			destTableName = j.Dest.Table
-		} else {
-			destTableName = alterJob.TableName
+		// before drop table, drop related view firstly
+		if !allViewDeleted {
+			var retryTimes int = 0
```
Comment (w41ter) [inline pkg/ccr/job.go]:
```suggestion
```

File: pkg/ccr/job.go [hunk 6]
```diff
@@ -1034,14 +1034,42 @@ func (j *Job) handleAlterJob(binlog *festruct.TBinlog) error {
...
+			views, err := j.IDest.GetAllViewsFromTable(destTableName)
+			if err != nil {
+				log.Errorf("when alter job, get view from table failed, err : %v", err)
+				if retryTimes < 3 {
+					retryTimes++
+					continue
+				}
```
Comment (w41ter) [inline pkg/ccr/job.go]:
```suggestion
				continue
```

**Pull request merged**